src/outputs/output_utils.hpp

```diff
@@ -200,6 +203,59 @@ struct AllSwarmInfo {
                bool is_restart);
 };
 
+template <typename T, typename Function_t>
+std::vector<T> FlattenBlockInfo(Mesh *pm, int shape, Function_t f) {
+  const int num_blocks_local = static_cast<int>(pm->block_list.size());
+  std::vector<T> data(shape * num_blocks_local);
+  int i = 0;
+  for (auto &pmb : pm->block_list) {
+    f(pmb.get(), data, i);
+  }
+  return data;
+}
+
+template <typename Data_t, typename idx_t, typename Function_t>
+void PackOrUnpackVar(MeshBlock *pmb, Variable<Real> *pvar, bool do_ghosts, idx_t &idx,
+                     std::vector<Data_t> &data, Function_t f) {
+  auto v_h = pvar->data.GetHostMirrorAndCopy();
```

Comment:
Just double checking (as host memory might actually be a constrained in some setups): Calling `GetHostMirrorAndCopy` from a `GetHostMirror` variable does *not* duplicate the memory allocation, correct?

Reply:
I'm not sure I understand the question---you mean in the case where the memory space and host space are the same right? `GetHostMirrorAndCopy` is a thin wrapper (with our data structures) around `Kokkos::create_mirror_view_and_copy`, which only allocates new memory if the source and target memory spaces are different.

Reply:
https://kokkos.github.io/kokkos-core-wiki/API/core/view/create_mirror.html?highlight=create_mirror_view_and_copy#_CPPv4I0DpE27create_mirror_view_and_copy14ImplMirrorTypeRKN4Impl12ViewCtorPropIDp12ViewCtorArgsEERK8ViewType

Reply:
That's what I thought, which is why I was double checking (as currently there's another `GetHostMirror` right before calling this function is being called).